src/sys/signalfd.rs

```diff
@@ -46,13 +44,14 @@ pub const SIGNALFD_SIGINFO_SIZE: usize = mem::size_of::<siginfo>();
 /// signalfd (the default handler will be invoked instead).
 ///
 /// See [the signalfd man page for more information](https://man7.org/linux/man-pages/man2/signalfd.2.html)
-pub fn signalfd(fd: RawFd, mask: &SigSet, flags: SfdFlags) -> Result<RawFd> {
+pub fn signalfd<F: AsFd>(fd: Option<F>, mask: &SigSet, flags: SfdFlags) -> Result<OwnedFd> {
```

Comment:
`pub fn signalfd<F: AsFd>(fd: Option<&F>` probably is the better way to go, the interface exposed from the `stdlib` [is not that great](https://github.com/rust-lang/rust/issues/88989#issuecomment-1035416012), sorry for my previous suggestion on changing the type to `<F: AsFd>(fd: F)`

Reply:
Given https://github.com/nix-rust/nix/pull/1928#discussion_r1043304050 I think we can resolve this?